packages/lit-dev-tools-esm/src/build-unversioned-docs.ts

```diff
@@ -0,0 +1,154 @@
+  CONTENT_PKG,
+  'site',
+  'docs',
+  SITE_LATEST_VERSION
+);
+console.log(LATEST_VERSION_CONTENT);
+const UNVERSIONED_VERSION_LOCATION = pathlib.resolve(
+  CONTENT_PKG,
+  'site',
+  'docs',
+  'unversioned'
+);
+
+/**
+ * This script builds our unversioned latest documentation for lit.dev. It
+ * locates the current latestVersion of Lit to show from `site.json`, then
+ * copies that documentation from the versioned subdirectory into an unversioned
+ * subdirectory.
+ *
+ * The following transforms are then applied on the files:
+ *   - Permalink frontmatter is added to strip the version from the URL. E.g.,
+ *     /docs/v2/* becomes /docs/*.
+ *   - Versioned cross links are detected and made unversioned.
+ */
+const buildAndTransformUnverionedDocs = async () => {
+  const walk = async (dirPath: string): Promise<void> => {
+    const entries = await fsPromise.readdir(dirPath, {withFileTypes: true});
+    await Promise.all(
+      entries.map(async (entry) => {
+        const childPath = pathlib.join(dirPath, entry.name);
+        if (entry.isDirectory()) {
+          return walk(childPath);
+        } else {
+          transformFile(childPath);
+        }
+      })
+    );
+  };
+
+  console.log('Starting build-unversioned-docs.js script');
+  await walk(LATEST_VERSION_CONTENT);
+  console.log('Completed build-unversioned-docs.js script');
+};
+
+/**
+ * Transform the given versioned content file and copy it to the unversioned
+ * directory.
+ */
+function transformFile(path: string) {
+  let fileContents = fs.readFileSync(path, {encoding: 'utf8'});
+  const relativeChildPath = pathlib.relative(LATEST_VERSION_CONTENT, path);
+  const ext = pathlib.extname(relativeChildPath);
+  let unversionedLocation = pathlib.join(
+    UNVERSIONED_VERSION_LOCATION,
+    relativeChildPath
+  );
+
+  if (ext === '.md') {
+    const fileName = pathlib.basename(unversionedLocation, '.md');
+    const [frontMatterData, restOfFile] = getFrontMatterData(fileContents);
+    const existingPermalink = frontMatterData.findIndex((val) =>
+      val.includes('permalink:')
+    );
+    if (existingPermalink !== -1) {
+      throw new Error(
+        'Unhandled case: Handle this by transforming the permalink here.'
+      );
+    }
+    if (fileName === 'index') {
+      frontMatterData.push(
+        `permalink: docs/${relativeChildPath.slice(0, -3)}.html`
+      );
+    } else {
+      frontMatterData.push(
+        `permalink: docs/${relativeChildPath.slice(0, -3)}/index.html`
+      );
+    }
+
+    fileContents = writeFrontMatter(frontMatterData) + restOfFile;
+  } else if (ext === '.json') {
+    if (
+      pathlib.basename(unversionedLocation, '.json') === SITE_LATEST_VERSION
+    ) {
+      unversionedLocation = pathlib.join(
+        pathlib.dirname(unversionedLocation),
+        'unversioned.json'
+      );
+      fileContents = JSON.stringify({
+        collection: 'docs-unversioned',
+        latestVersion: SITE_LATEST_VERSION,
+      });
+    }
```

Comment:
does this need an else block to throw in case of unexpected .json file? alternatively, if it's a given that this'll be the only json file within this directory, maybe the if condition could be dropped?

a comment that this case is to handle `v[X]/v[X].json` file would be nice. personally, it was a bit disorienting to see this between the .md and .html handling when this is more of a one off case.

Reply:
Check out new code comments in the newest commit & let me know if it sufficiently addresses this. Great suggestion!